src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,171 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> currentNode;
+    private int size;
+
+    public MyLinkedList() {
+        this.head = null;
+        this.tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        currentNode = new Node<>(value);
+        if (size == 0) {
+            head = currentNode;
+        } else if (tail == null) {
+            tail = currentNode;
+            head.next = tail;
+            tail.prev = head;
+        } else {
+            Node<T> lastNode = tail;
+            tail = currentNode;
+            tail.prev = lastNode;
+            tail.prev.prev.next = lastNode;
+            lastNode.next = tail;
+        }
```

Comment:
If the size if not 0, it would be enough to just find the last element and set the `next` field with the new node